packages/components/src/popover/index.js

```diff
@@ -159,30 +196,29 @@ const Popover = (
 	}, [ anchorRef, anchorRect, getAnchorRect ] );
 
 	/**
-	 * Offsets the position of the popover when the anchor is inside an iframe.
+	 * Adjusts the position of the popover when the anchor is inside an iframe
+	 * so that the popover aligns correctly with its anchor.
 	 */
-	const frameOffset = useMemo( () => {
-		const { defaultView } = ownerDocument;
-		const { frameElement } = defaultView;
+	const iframeOffsetMiddleware = useMemo( () => {
+		const frameElement = ownerDocument?.defaultView?.frameElement;
 
 		if ( ! frameElement || ownerDocument === document ) {
 			return undefined;
 		}
 
-		const iframeRect = frameElement.getBoundingClientRect();
-		return {
-			name: 'iframeOffset',
-			fn( { x, y } ) {
-				return {
-					x: x + iframeRect.left,
-					y: y + iframeRect.top,
-				};
-			},
-		};
+		const { top, left } = frameElement.getBoundingClientRect();
+
+		return offsetMiddleware( ( { placement } ) => {
+			const computedMainAxis = getMainAxisFromPlacement( placement );
```

Comment:
The change here looks very similar to what I did in #42417 to fix the incorrect iframe offset.

Something to note is that floating ui's mainAxis isn't always the same. For `flip` and `offset` the main axis for `top` / `bottom` would be `y`, but for `shift` the main axis is `x`:
- shift - https://floating-ui.com/docs/shift#mainaxis
- flip - https://floating-ui.com/docs/flip#mainaxis

So I don't know that it's useful to have a general purpose `getMainAxisFromPlacement` function.

I only know this because I did the exact same thing in my PR 😄  I ended up inlining the logic. I think this is why floating-ui doesn't export a `getMainAxis` function.

Reply:
Interesting — I borrowed this util (and `getSide`) directly [from floating UI 's codebase ](https://github.com/floating-ui/floating-ui/blob/master/packages/core/src/utils/getMainAxisFromPlacement.ts), but I had completely overlooked this difference. I'll go ahead and inline the logic

Reply:
> I'll go ahead and inline the logic

As discussed in the other conversation, I'll actually stop working on this PR for a while until https://github.com/WordPress/gutenberg/pull/42417 is merged.

The bit of code being discussed in this conversation will be super-seeded by the changes in that PR.